### PR TITLE
remove TODO:parallized component  health checking

### PR DESCRIPTION
**Pull request by mqliang**
remove TODO:parallized component health checking

### CODE REVIEW

File: pkg/registry/componentstatus/rest.go [hunk 1]
```diff
@@ -53,11 +54,22 @@ func (rs *REST) NewList() runtime.Object {
...
-	reply := []api.ComponentStatus{}
+	wait := sync.WaitGroup{}
+	wait.Add(len(servers))
+	statuses := make(chan api.ComponentStatus, len(servers))
 	for name, server := range servers {
-		status := rs.getComponentStatus(name, server)
-		reply = append(reply, *status)
+		go func() {
+			defer wait.Done()
+			status := rs.getComponentStatus(name, server)
```
Comment (fabioy) [inline pkg/registry/componentstatus/rest.go]: This won't work. Due to Go's scoping rules for the "for loop" vars, the go function will not evaluate "name" and "server" in the way you want (I believe it'll only get the last entry). See https://play.golang.org/p/yRbY3uRb39 for an example.

Comment (mqliang) [inline pkg/registry/componentstatus/rest.go]: Thanks for pointing out this,fixed.

**Comment by fabioy** [comment]
Change looks good. Please squash the commits to a single one and I'll LGTM it. Thanks.

**Comment by mqliang** [comment]
@fabioy Squashed.

**Pull request merged**